server_http/static/main_script.js

```diff
@@ -434,22 +448,61 @@ function adjust_displayed_messages() {
 }
 
 
+function update_save_login() {
+    let id = generate_random_string(id_length);
+    let username = my_name_element.value + id;
+    localStorage.setItem(ACTIVE_USER_STORAGE, username);
+    return username;
+}
+
+
 function activate_actions_on_entering_chat() {
     let stored_user_name = localStorage.getItem(ACTIVE_USER_STORAGE);
     if (stored_user_name && my_name_element.value === retrieve_display_login(stored_user_name)) {
         login = stored_user_name;
         console.log('equal logins');
     } else {
-        var id = generate_unique_id(id_length);
-        login = my_name_element.value + id;
-        localStorage.setItem(ACTIVE_USER_STORAGE, login);
+        login = update_save_login();
     }
-    chat_change(chat_destination_element.value);
+    enter_chat(chat_destination_element.value)
+    connect(login, chat);
     add_chat(chat);
     console.log(login, chat);
 }
 
 
+function share_chat(platform) {
+    let chat_url = window.location.href;
+    let platform_share_url;
+    switch (platform) {
```

Comment:
it should be removed from this place as well, since this share functionality is not used anymore